dynalib/inc/dynalib.h

```diff
@@ -48,13 +48,38 @@
 
 #ifdef DYNALIB_EXPORT
 
+#if __cplusplus >= 201103 // C++11
+
+    #include <type_traits>
+
+    template<typename T1, typename T2>
+    constexpr const void* dynalib_check_function_type(T2 *func) {
+        static_assert(std::is_same<T1, T2>::value, "Signature of the dynamically exported function has changed");
+        return (const void*)func;
+    }
+    #define DYNALIB_FN(tablename,name,type) \
+        dynalib_check_function_type<type>(name),
+
+#elif __STDC_VERSION__ >= 199901 && !__STRICT_ANSI__ // C99 with GNU extensions
+
+    #define DYNALIB_FN(tablename,name,type) \
+        __builtin_choose_expr(__builtin_types_compatible_p(type, __typeof__(name)), name, sizeof(struct { \
```

Comment:
could you replace the 2nd argument `name` with `(const void*)&name`, just so that we are certain there is no code change between this and the previous version?  (That's how name was used in DYNALIB_FN previously.)